assembl/static2/js/app/components/debate/common/topPostFormContainer.jsx

```diff
@@ -0,0 +1,95 @@
+// @flow
+import React from 'react';
+import { Grid, Row, Col } from 'react-bootstrap';
+import { Translate } from 'react-redux-i18n';
+import classNames from 'classnames';
+import TopPostForm from './topPostForm';
+import { hexToRgb } from '../../../utils/globalFunctions';
+import { MIN_WIDTH_COLUMN } from '../../../constants';
+
+class TopPostFormContainer extends React.Component {
+  getClassNames() {
+    const { messageColumns, isColumnViewInline } = this.props;
+    return classNames({ 'columns-view': messageColumns.length > 1 }, { 'columns-view-inline': isColumnViewInline });
+  }
+  getColumnsInfos() {
+    const { messageColumns } = this.props;
+    let columnsInfos = [];
+    if (messageColumns.length > 1) {
+      columnsInfos = messageColumns;
+    } else {
+      columnsInfos.push({ messageClassifier: '', color: '', name: '' });
+    }
+
+    return columnsInfos;
+  }
+  render() {
+    const { ideaId, refetchIdea, messageColumns, isColumnViewInline } = this.props;
+    const columnsInfos = this.getColumnsInfos();
+    return (
+      <Grid fluid className={messageColumns.length > 1 ? '' : 'background-color'}>
+        <div className="max-container">
+          <Row>
+            <div className={this.getClassNames()}>
+              {columnsInfos.map((column) => {
```

Comment:
Because the high level container is `pages/idea.jsx`. This component could be display as a thread or as a multi column view. So the multi column is not a high level component with its own route.

Reply:
Yup, that's clear. The highest element is the `pages/index.jsx`, which does a check if there are messageColumns. Inside this component, there should be a check, if it's n-column, display a whole new component, else display what it currently displayed (as a component). 

The n-column component then would have a hierarchy kind of like this:
```
<NColumns>
 {columns.map(col => { 
    return (
      <TopPost {...propsForColumns} />
      <Tree {...propsForColumnTree} />
    );
  })}
</NColumns>
```

That's how I thought the architecture of the front-end would look like. Instead, it is more like:

```
// In TopPostContainer.jsx
{columns.map(col => {
  return (<TopPostColumn {...propsForColumn} />);
})}

// In another file
{columns.map(col => {
  return (<Tree {...propsForColumn} />);
})}
```

It's a choice. I wondered why you chose to filter on the columns array twice across different files, whereas you could have made a more clear component?

Also this will make isolated testing the component a bit more cumbersome.

Reply:
Yes, your approach is the best... I should refactor it. I need 4h to do that.